models/user.js

```diff
@@ -296,56 +297,104 @@ UserService.createLocalUsers = (users) => {
 };
 
 /**
+ * Check the requested displayname for naughty words (currently in English) and special chars
+ * @param  {String}   displayName word to be checked for profanity
+ * @return {Promise}  rejected if the machine's sensibilites are offended
+ */
+const isValidDisplayName = (displayName) => {
+  const onlyLettersNumbersUnderscore = /^[a-z0-9_]+$/;
+
+  if (!displayName) {
+    const ErrMissingDisplay = new Error('DISPLAY_NAME_REQUIRED');
+    ErrMissingDisplay.status = 400;
+    return Promise.reject(ErrMissingDisplay);
+  }
+
+  if (!onlyLettersNumbersUnderscore.test(displayName)) {
+    const ErrSpecialChars = new Error('NO_SPECIAL_CHARACTERS');
+    ErrSpecialChars.status = 400;
+    return Promise.reject(ErrSpecialChars);
+  }
+
+  const hasBadWords = naughtyWords.some(badWord => {
+
+    // test the word AFTER the interspersed _ characters are removed.
+    return new RegExp(badWord, 'ig').test(displayName.replace(/_/g, ''));
+  });
+
+  if (hasBadWords) {
+    const ErrProfanity = new Error('PROFANITY_ERROR');
+    ErrProfanity.status = 400;
+    return Promise.reject(ErrProfanity);
+  } else {
+    return Promise.resolve(displayName);
+  }
+};
+
+/**
  * Creates the local user with a given email, password, and name.
  * @param  {String}   email       email of the new user
  * @param  {String}   password    plaintext password of the new user
  * @param  {String}   displayName name of the display user
  * @param  {Function} done        callback
  */
 UserService.createLocalUser = (email, password, displayName) => {
+
   if (!email) {
-    return Promise.reject('email is required');
+    const ErrMissingEmail = new Error('EMAIL_REQUIRED');
```

Comment:
so before, we were just throwing away the `message` part of the error. I updated the response helper on the frontend to pass the message through. It's a string constant like that so that I can look up the code and translate it into Spanish or whatever.